xbmc/addons/Repository.h

```diff
@@ -20,23 +20,23 @@ namespace ADDON
 {
 class CAddonExtensions;
 
-class CRepository : public CAddon
+struct RepositoryDirInfo
 {
-public:
-  struct DirInfo
-  {
-    AddonVersion minversion{""};
-    AddonVersion maxversion{""};
-    std::string info;
-    std::string checksum;
-    KODI::UTILITY::CDigest::Type checksumType{KODI::UTILITY::CDigest::Type::INVALID};
-    std::string datadir;
-    std::string artdir;
-    KODI::UTILITY::CDigest::Type hashType{KODI::UTILITY::CDigest::Type::INVALID};
-  };
+  AddonVersion minversion{""};
+  AddonVersion maxversion{""};
+  std::string info;
+  std::string checksum;
+  KODI::UTILITY::CDigest::Type checksumType{KODI::UTILITY::CDigest::Type::INVALID};
+  std::string datadir;
+  std::string artdir;
+  KODI::UTILITY::CDigest::Type hashType{KODI::UTILITY::CDigest::Type::INVALID};
+};
 
-  typedef std::vector<DirInfo> DirList;
+typedef std::vector<RepositoryDirInfo> RepositoryDirList;
```

Comment:
This can be changed to:

```c++
using RepositoryDirList = std::vector<RepositoryDirInfo>;
```

Reply:
There are many other typedefs used in the addons component. I suggest to change them in another PR.